### PR TITLE
Add more huggingface models in benchmark tools

### PR DESCRIPTION
**Pull request by wangyems**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 1]
```diff
@@ -15,6 +15,33 @@
 
 logger = logging.getLogger(__name__)
 
+torch_func = {"triu": torch.triu }
+
+# reference: https://github.com/microsoft/onnxruntime/commit/ed67af8977c5c5068b9d2d1d30bc08db706ff80a#diff-f8610a9125f273a1ab46a062ee263a63
+def triu_onnx(x, diagonal=0, out=None):
+    assert out is None
+    assert len(x.shape) == 2 and x.size(0) == x.size(1)
+    """
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:24]: commented code can be removed.

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 2]
```diff
@@ -15,6 +15,33 @@
 
 logger = logging.getLogger(__name__)
 
+torch_func = {"triu": torch.triu }
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:18]: format the scripts

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 3]
```diff
@@ -15,6 +15,33 @@
...
+
+# reference: https://github.com/microsoft/onnxruntime/commit/ed67af8977c5c5068b9d2d1d30bc08db706ff80a#diff-f8610a9125f273a1ab46a062ee263a63
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:20]: Add some useful comments here instead of a link to another commit.

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 4]
```diff
@@ -243,4 +272,8 @@ def export_onnx_model(model_name, opset_version, use_external_data_format, model
                                                 use_external_data_format)
             optimize_onnx_model_by_ort(onnx_model_path, ort_model_path, use_gpu, overwrite, model_fusion_statistics)
 
-    return onnx_model_path, is_valid_onnx_model, config.vocab_size, tokenizer.max_model_input_sizes[model_name]
+    model_name_in_tokenizer = model_name
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:275]: instead of doing the tricky conversion, how about 
```
    max_input_sizes = tokenizer.max_model_input_sizes[model_name] if model_name in tokenizer.max_model_input_sizes else 1024
```

File: onnxruntime/python/tools/transformers/huggingface_models.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    "bert-large-cased-whole-word-masking-finetuned-squad": (["input_ids", "attention_mask", "token_type_ids"], 11, False, "bert"),
+    "bert-base-cased-finetuned-mrpc": (["input_ids", "attention_mask", "token_type_ids"], 11, False, "bert"),
+    "bert-base-german-dbmdz-cased": (["input_ids", "attention_mask", "token_type_ids"], 11, False, "bert"),
+    "bert-base-german-dbmdz-uncased": (["input_ids", "attention_mask", "token_type_ids"], 11, False, "bert"),
+    # todo: more models to add
+# GPT
+    "openai-gpt": (["input_ids"], 11, False, "gpt2"),  # no past state inputs
+# GPT-2
+    "gpt2": (["input_ids"], 11, False, "gpt2"),  # no past state inputs & outputs
+    "gpt2-medium": (["input_ids"], 11, True, "gpt2"),
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/huggingface_models.py:31]: nit: gpt2-medium do not need external data format since model < 2GB.

File: onnxruntime/python/tools/transformers/huggingface_models.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+# Copyright (c) Microsoft Corporation.  All rights reserved.
+# Licensed under the MIT License.  See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+# List of pretrained models: https://huggingface.co/transformers/pretrained_models.html
+# Pretrained model name to a tuple of input names, opset_version, use_external_data_format and optimization model type
+MODELS = {
+# BERT
+    "bert-base-uncased": (["input_ids", "attention_mask", "token_type_ids"], 11, False, "bert"),
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/huggingface_models.py:11]: nit: no much value to add so many bert models that quite similar. 

I think the following are typical:
bert-base-cased, bert-large-cased, bert-large-cased-whole-word-masking-finetuned-squad, bert-base-cased-finetuned-mrpc

File: onnxruntime/python/tools/transformers/huggingface_models.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+# RoBERTa
+    "roberta-base": (["input_ids", "attention_mask"], 11, False, "bert"),
+    "roberta-large": (["input_ids", "attention_mask"], 11, False, "bert"),
+    "roberta-large-mnli": (["input_ids", "attention_mask"], 11, False, "bert"),
+    "distilroberta-base": (["input_ids", "attention_mask"], 11, False, "bert"),
+    "roberta-base-openai-detector": (["input_ids", "attention_mask"], 11, False, "bert"),
+    "roberta-large-openai-detector": (["input_ids", "attention_mask"], 11, False, "bert"),
+# DistilBERT
+    "distilbert-base-uncased": (["input_ids", "attention_mask"], 11, False, "bert"),
+    "distilbert-base-uncased-distilled-squad": (["input_ids", "attention_mask"], 11, False, "bert"),  # no past state inputs & outputs
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/huggingface_models.py:59]: Update the comment for distilbert models.

File: onnxruntime/python/tools/transformers/huggingface_models.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    "distilbert-base-cased": (["input_ids", "attention_mask"], 11, False, "bert"),  # no past state inputs & outputs
+    "distilbert-base-cased-distilled-squad": (["input_ids", "attention_mask"], 11, False, "bert"),  # no past state inputs & outputs
+    "distilgpt2": (["input_ids"], 11, False, "gpt2"),  # no past state inputs & outputs
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/huggingface_models.py:62]: distilgpt2 should be another section (not belongs to DISTILBERT)

File: onnxruntime/python/tools/transformers/huggingface_models.py (hunk 6 above)
Comment (wangyems) [reply onnxruntime/python/tools/transformers/huggingface_models.py:11]: Verified that exported squad model has two outputs

**Review by tianleiwu — APPROVED** [review]

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by thiagocrepaldi** [comment]
https://github.com/microsoft/onnxruntime/pull/5002 will fix this code owner issue, but it is pending on CI for some time now
